extensions/wikia/HTTPSSupport/tests/HTTPSSupportHooksTest.php

```diff
@@ -86,4 +90,98 @@ public function onBeforeResourceLoaderCSSMinifierDataProvider() {
 			]
 		];
 	}
+	/**
+	 * @param string $url
+	 * @param string $environment
+	 * @param string $expectedResult
+	 *
+	 * @dataProvider onLinkerMakeExternalLinkDataProvider
+	 */
+	public function testOnLinkerMakeExternalLink( $url, $environment, $expectedResult ) {
+
+		$exampleString = '';
+		$exampleBool = false;
+		$exampleArray = [];
+
+		$this->mockEnvironment( $environment );
+		HTTPSSupportHooks::onLinkerMakeExternalLink($url, $exampleString, $exampleBool, $exampleArray);
+		$this->unsetGlobals();
+		$this->assertEquals( $expectedResult, $url );
+
+	}
+
+	public function onLinkerMakeExternalLinkDataProvider() {
+		return [
+			[
+				'$url' => 'http://www.example.com',
```

Comment:
are you sure about using array keys for the argument? I have not seen that in other test data providers?

Reply:
I think I've seen it in MercuryApi tests for example: https://github.com/Wikia/app/blob/44510b86471c2fc1c6b799f9599746da1dd1cbee/extensions/wikia/MercuryApi/tests/MercuryApiModelTest.php#L36 or https://github.com/Wikia/app/blob/44510b86471c2fc1c6b799f9599746da1dd1cbee/extensions/wikia/MercuryApi/tests/MercuryApiModelTest.php#L244

Reply:
Interesting, I don't see it being mentioned in https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers